tests/Functional/ConnectionTest.php

```diff
@@ -117,6 +117,13 @@ public function testTransactionNestingLevelIsResetOnReconnect(): void
         $connection->rollBack();
 
         self::assertEquals(0, $connection->fetchOne('select count(*) from test_nesting'));
+
+        if ($sqlitePath === null) {
+            return;
+        }
+
+        $connection->close();
+        unlink($sqlitePath);
```

Comment:
The teardown routines should stay where they are. This code will not be executed if the above assertion fails.

Reply:
the teardown is specific for one test only, fixed using try/finally block, this fixes also the filepath defined on multiple places